### PR TITLE
Use `getExecutionEngine` to obtain the JIT for Cling > LLVM16

### PR DESCRIPTION
**Pull request by aaronj0**
To also be backported to the root-v1.0 tag

### CODE REVIEW

File: lib/Interpreter/Compatibility.h [hunk 1]
```diff
@@ -85,17 +85,35 @@ inline void maybeMangleDeclName(const clang::GlobalDecl& GD,
...
+#endif
+
+// Note: The offsets changed in LLVM 16 with the introduction of
+#if CLANG_VERSION_MAJOR == 16
+#ifdef __APPLE__
+  const unsigned m_ExecutorOffset = 62;
+#else
+  const unsigned m_ExecutorOffset = 72;
+#endif // __APPLE__
+#endif
```
Comment (vgvassilev) [inline lib/Interpreter/Compatibility.h:115]: How is this block different from the previous?

Comment (aaronj0) [reply lib/Interpreter/Compatibility.h:115]: The offsets have not yet been updated for cling/llvm 16, will do

Comment (aaronj0) [reply lib/Interpreter/Compatibility.h]: @vgvassilev done

**Review by vgvassilev — APPROVED** [review]
Lgtm!

**Pull request merged**